staging/src/k8s.io/apimachinery/pkg/runtime/schema/group_version.go

```diff
@@ -303,3 +305,40 @@ func FromAPIVersionAndKind(apiVersion, kind string) GroupVersionKind {
 	}
 	return GroupVersionKind{Kind: kind}
 }
+
+// APILifecycle stores information about the versions when a group version or a resource type is introduced and removed.
+// It is checked to make sure a gvr is not available before its type is introduced or after it is removed.
+// This can be set when registering the lifecycle of a group version with SetGroupVersionLifecycle or
+// when registering the lifecycle of a resource type with SetResourceLifecycle in a Scheme.
+// To set the APILifecycle for your resources:
+// in register.go,
+//
+// 	func addKnownTypes(scheme *runtime.Scheme) error {
+// 		scheme.AddKnownTypes(SchemeGroupVersion,
+// 			&Example{},
+// 			&ExampleList{},
+// 		)
+// 		metav1.AddToGroupVersion(scheme, SchemeGroupVersion)
+// 		// If all individual resource types of this group share the lifecycle of the group:
+// 		scheme.SetGroupVersionLifecycle(SchemeGroupVersion, schema.APILifecycle{
+// 			IntroducedVersion: version.MajorMinor(1, 20),
+// 			RemovedVersion:    version.MajorMinor(1, 26),
+// 		}
```

Comment:
It seems like our intention is to unconditionally tag EVERY resource with `k8s:prerelease-lifecycle-gen`. In that case every resource should have its own lifecycle and I am not sure a distinct GroupVersion lifecycle concept is worth the additional complexity

Reply:
Caught up with some prior discussion. IMO the group-version fallback wouldnt be necessary if we can auto-generate the resource lifecycle registration https://github.com/kubernetes/kubernetes/pull/122891#discussion_r1505194247